zingolib/src/wallet/transaction_records_by_id/trait_inputsource.rs

```diff
@@ -126,31 +124,40 @@ impl InputSource for TransactionRecordsById {
         anchor_height: zcash_primitives::consensus::BlockHeight,
         exclude: &[Self::NoteRef],
     ) -> Result<SpendableNotes<Self::NoteRef>, Self::Error> {
-        let mut unselected =
-            self.get_spendable_note_ids_and_values(sources, anchor_height, exclude);
-
+        let mut unselected = self
+            .get_spendable_note_ids_and_values(sources, anchor_height, exclude)
+            .into_iter()
+            .map(|(id, value)| NonNegativeAmount::from_u64(value).map(|value| (id, value)))
+            .collect::<Result<Vec<_>, _>>()
+            .map_err(InputSourceError::InvalidValue)?;
         unselected.sort_by_key(|(_id, value)| *value);
-
         let mut selected = vec![];
-
         let mut index_of_unselected = 0;
+
         loop {
+            // if no unselected notes are available, return the currently selected notes even if the target value has not been reached
```

Comment:
`If there are no unselected notes as of this iteration...`

Reply:
im happy for you to open a PR to make this change, i was just making @fluidvanadium 's comment more correct